src/pages/Learn/Learn.jsx

```diff
@@ -15,42 +16,68 @@ const Learn = () => {
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(null);
 
-  useEffect(() => {
-    const fetchCourseData = async () => {
-      try {
-        if (!courseId) {
-          throw new Error('Course ID not found. Please try again.');
-        }
-
-        const courseData = await fetchCourseByCourseId(courseId);
-        const lessonsData = await fetchCourseLessons(courseId);
-        setCourse(courseData);
-        setLessons(lessonsData);
-
-        // Auto-select first lesson or the one in the URL
-        const lessonInUrl = lessonsData.find(
-          (lesson) =>
-            lesson.title ===
-            decodeURIComponent(location.pathname.split('/').pop()),
-        );
-        setSelectedLesson(lessonInUrl || lessonsData[0]);
-        setLoading(false);
-      } catch (error) {
-        setError(error.message || 'Error fetching course lessons');
-        console.error('Error fetching course lessons:', error);
-        setLoading(false);
+  // State for controlling which component to display: 'lesson', 'feedback', or 'courseInfo'
+  const [view, setView] = useState('lesson');
+
+  const [isLessonListExpanded, setIsLessonListExpanded] = useState(true);
+
+  const fetchCourseData = async () => {
+    try {
+      if (!courseId) {
+        throw new Error('Course ID not found. Please try again.');
       }
-    };
 
-    fetchCourseData();
-  }, [courseId, location.pathname]);
+      const courseData = await fetchCourseByCourseId(courseId);
+      const lessonsData = await fetchCourseLessons(courseId);
+      setCourse(courseData);
+      setLessons(lessonsData || []);
+
+      // Auto-select first lesson or the one in the URL
+      const lessonInUrl = lessonsData.find(
+        (lesson) =>
+          lesson.title ===
+          decodeURIComponent(location.pathname.split('/').pop())
+      );
+      setSelectedLesson(lessonInUrl || lessonsData[0]);
+      setLoading(false);
+    } catch (error) {
+      setError(error.message || 'Error fetching course lessons');
+      console.error('Error fetching course lessons:', error);
+      setLoading(false);
+      navigate('/');
+    }
+  };
+
+  useEffect(() => {
+    if (!courseId) {
+      navigate('/'); // Redirect to a different page if courseId is not available
```

Comment:
I would say if a course and its lesson are not loading, redirect them to the user home page where they can search for other courses available. Just my opinion but its up to you of course.

Reply:
I will look into it!